packages/multichain/src/caip25Permission.ts

```diff
@@ -137,21 +137,26 @@ const specificationBuilder: PermissionSpecificationBuilder<
         }
       };
 
-      assertScopesSupported(normalizedRequiredScopes, {
-        isChainIdSupported,
-      });
-      assertScopesSupported(normalizedOptionalScopes, {
-        isChainIdSupported,
-      });
+      const allRequiredScopesSupported = Object.keys(requiredScopes).every(
+        (scopeString) =>
+          isSupportedScopeString(scopeString, isChainIdSupported),
+      );
+      const allOptionalScopesSupported = Object.keys(optionalScopes).every(
+        (scopeString) =>
+          isSupportedScopeString(scopeString, isChainIdSupported),
+      );
+      if (!allRequiredScopesSupported || !allOptionalScopesSupported) {
+        throw Caip25Errors.requestedChainsNotSupportedError();
```

Comment:
Is this the error we want to throw? If we've reached here wouldn't it kindof be an internal error since we should have filtered for supported chains prior to this step?

Reply:
fixed here https://github.com/MetaMask/core/pull/4950/commits/0e623f0332ef7f85530fbd2dce6971cbfeb961da